### PR TITLE
edit documentation 

### PR DESCRIPTION
**Pull request by anneleong**
edit developers guide, following closely to https://se-education.org/addressbook-level4/DeveloperGuide.html

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -703,55 +745,62 @@ The flow can be seen from the sequence diagram below:
...
+- wants to be able to save news articles that they want to read later on.
+- wants personalised suggestions on news articles to read.
+- prefers typing over mouse input.
+- is reasonably comfortable using CLI apps.
+
+<h3 id = "value-proposition"> Value Proposition</h3>
+Allows the exploration of a range of articles from a variety of sources faster than browsing the internet and various 
+news sources for articles.
+
+<h3 id="user-stories"> Appendix B: User Stories </h3>
```
Comment (BenjoBurger) [inline docs/DeveloperGuide.md:762]: This should use the h2 header instead!

File: docs/UML_Diagrams/filterFunctionSequence.png
Comment (BenjoBurger) [inline docs/UML_Diagrams/filterFunctionSequence.png:1]: wat happen to the source function

Comment (anneleong) [reply docs/UML_Diagrams/filterFunctionSequence.png:1]: sorry i dont thin my sequence diagram has a source function

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (anneleong) [reply docs/DeveloperGuide.md:763]: thank u :)

**Review by HiteshriAcharya — APPROVED** [review]

**Pull request merged**